test/conf_include_test.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <string.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP '\\'
+# define CHDIR _chdir
+#else
+# include <unistd.h>
+# define DIRSEP '/'
+# define CHDIR chdir
+#endif
+
+/* changes path to that of the filename */
+static int change_path(const char *file)
+{
+    char *s = OPENSSL_strdup(file);
+    char *p;
+    int ret;
+
+    if (s == NULL)
+        return -1;
+
+    p = strrchr(s, DIRSEP);
+    if (p == NULL)
+        return 0;
+    *p = 0;
+
+    fprintf(stderr, "changing path to %s\n", s);
+    ret = CHDIR(s);
+    OPENSSL_free(s);
+    return ret;
+}
+
+/* This test program checks the operation of the .include directive.
+ */
+
+static BIO *dup_bio_in(void)
+{
+    return BIO_new_fp(stdin, BIO_NOCLOSE | BIO_FP_TEXT);
+}
+
+static BIO *dup_bio_out(void)
+{
+    BIO *b = BIO_new_fp(stdout, BIO_NOCLOSE | BIO_FP_TEXT);
+    return b;
+}
+
+static BIO *bio_open_default_(const char *filename, const char *mode, int quiet)
+{
+    BIO *ret;
+
+    if (filename == NULL || strcmp(filename, "-") == 0) {
+        ret = *mode == 'r' ? dup_bio_in() : dup_bio_out();
+        if (quiet) {
+            ERR_clear_error();
+            return ret;
+        }
+        if (ret != NULL)
+            return ret;
+        fprintf(stderr,
+                   "Can't open %s, %s\n",
+                   *mode == 'r' ? "stdin" : "stdout", strerror(errno));
```

Comment:
Two things here. Styling nit. That's not how we format function calls. And secondly [at more importantly], I recall discussion about not using fprintf in [new] test programs. I think test_printf_stderr is the one to use here. But I'd like to hear somebody confirming it...

Reply:
Yes, use test_printf_stderr not fprintf.